src/flib/spio_init.F90

```diff
@@ -0,0 +1,655 @@
+    INTEGER(C_INT) :: cerr
+
+#ifdef TIMING
+    ! Ignore timing since the call is not costly
+    ! call t_startf("PIO:set_hint")
+#endif
+
+    ! The C versions of the strings need an extra char for C_NULL_CHAR
+    chint_sz = LEN_TRIM(hint) + 1
+    chval_sz = LEN_TRIM(hval) + 1
+
+    ALLOCATE(chint(chint_sz), chval(chval_sz))
+
+    ! Convert the hint & hint value from Fortran strings to C strings
+    lerr = f2cstring(iosys, TRIM(hint), chint, chint_sz, chint_sz,&
+                      cstr_add_null = .TRUE.)
+    IF(lerr /= PIO_NOERR) THEN
+      IF(PRESENT(ierr)) THEN
+        ierr = lerr
+      END IF
+      WRITE(log_msg, *) "Setting hint failed. Converting the hint name ",&
+                        "to a C string failed (hint =", TRIM(hint),&
+                        ", value =", TRIM(hval), ")"
+      lerr = pio_error(iosys, lerr, __PIO_FILE__, __LINE__, TRIM(log_msg))
+      RETURN
+    END IF
+
+    lerr = f2cstring(iosys, TRIM(hval), chval, chval_sz, chval_sz,&
+                      cstr_add_null = .TRUE.)
+    IF(lerr /= PIO_NOERR) THEN
+      IF(PRESENT(ierr)) THEN
+        ierr = lerr
+      END IF
+      WRITE(log_msg, *) "Setting hint failed. Converting the hint value ",&
+                        "to a C string failed (hint =", TRIM(hint),&
+                        ", value =", TRIM(hval), ")"
+      lerr = pio_error(iosys, lerr, __PIO_FILE__, __LINE__, TRIM(log_msg))
+      RETURN
+    END IF
+
+    cerr = PIOc_set_hint(iosys%iosysid, chint, chval)
+    IF(PRESENT(ierr)) THEN
+      ierr = INT(cerr)
+    ENDIF
+
+    DEALLOCATE(chint, chval)
+
+#ifdef TIMING
+    ! Ignore timing since the call is not costly
+    !call t_stopf("PIO:set_hint")
+#endif
+  END SUBROUTINE pio_set_hint
+
+!>
+!! @public
+!! @defgroup PIO_set_rearr_opts PIO_set_rearr_opts
+!! @brief Set the I/O data rerranger options for an I/O system
+!!
+!! @details
+!! @param[inout] iosys The handle to I/O system
+!! @param[in] comm_type @copydoc PIO_rearr_comm_t
+!! @param[in] fcd @copydoc PIO_rearr_comm_dir
+!! @param[in] enable_hs_c2i Enable handshake (compute procs to I/O procs)
+!! @param[in] enable_isend_c2i  Enable isends (compute procs to I/O procs)
+!! @param[in] max_pend_req_c2i  Maximum pending requests (compute procs to I/O procs)
+!! @param[in] enable_hs_i2c Enable handshake (I/O procs to compute procs)
+!! @param[in] enable_isend_i2c  Enable isends (I/O procs to compute procs)
+!! @param[in] max_pend_req_i2c  Maximum pending requests (I/O procs to compute procs)
+!! @returns @copydoc error_code
+!!
+!! @copydoc PIO_rearr_comm_fc_options
+!!
+  INTEGER FUNCTION pio_set_rearr_opts(iosys, comm_type, fcd,&
+                                      enable_hs_c2i, enable_isend_c2i,&
+                                      max_pend_req_c2i,&
+                                      enable_hs_i2c, enable_isend_i2c,&
+                                      max_pend_req_i2c) RESULT(ierr)
+
+    TYPE(iosystem_desc_t), INTENT(INOUT) :: iosys
```

Comment:
(IN) instead of (INOUT)?

Reply:
This is historic (scorpio classic changed the iosys), so will leave it as is